File: src/main/java/lk/tech/tgcontrollerclient/services/commands/impl/CommandShutdown.java
package lk.tech.tgcontrollerclient.services.commands.impl;

import lk.tech.tgcontrollerclient.dto.Result;
import lk.tech.tgcontrollerclient.services.commands.AbstractCommand;

public class CommandShutdown extends AbstractCommand {

    @Override
    public Result run(String command) {
//        ProcessBuilder pb = new ProcessBuilder("shutdown", "/s", "/f", "/t", "0");
//        try {
//            pb.start();
//            return new Result("Success");
//        } catch (IOException e) {
//            return new Result("Fail");
//        }
        return new Result("Success");
    }

    @Override
    public String condition() {
        return "/shutdown";
    }
}

File: services/commands/command_shutdown.go
package commands

import (
	"tg-controller-client/dto"
)

// CommandShutdown — аналог Java CommandShutdown
type CommandShutdown struct {
	BaseCommand
}

func (c *CommandShutdown) Condition() string {
	return "/shutdown"
}

func (c *CommandShutdown) Run(cmd string) dto.ResultInterface {

	// --- если хочешь настоящий shutdown Windows ---
	// out := exec.Command("shutdown", "/s", "/f", "/t", "0").Start()
	// if out != nil {
	//     return dto.Result{Status: "Fail"}
	// }
	// return dto.Result{Status: "Success"}

	// Пока — как в твоём Java-коде:
	return dto.Result{Status: "Success"}
}

func (c *CommandShutdown) Analyze(cmd string) dto.ResultInterface {
	return c.BaseCommand.Analyze(cmd, c)
}
